Lib/test/test_traceback.py

```diff
@@ -92,6 +92,28 @@ def test_base_exception(self):
         lst = traceback.format_exception_only(e.__class__, e)
         self.assertEqual(lst, ['KeyboardInterrupt\n'])
 
+    def test_traceback_recursionerror(self):
+        # Test that for long traceback chains traceback does not itself
+        # raise a recursion error while printing (Issue43048)
+
+        # Calling f() creates a stack-overflowing __context__ chain.
+        def f():
+            try:
+                raise ValueError('hello')
+            except ValueError:
+                f()
+
+        try:
+            f()
+        except RecursionError as e:
+            # Creating this exception should not fail if internally
+            # the traceback exception is constructed safely with respect
+            # to recursion context (since TracebackExceptions are used
+            # for printing and formatting exceptions we should allow their
+            # creation even for RecursionErrors with long chains).
+            exc_info = sys.exc_info()
+            traceback.TracebackException(exc_info[0], exc_info[1], exc_info[2])
```

Comment:
The test needs to call format() to make sure the recursion there can handle the truncated chain (it might have more function calls per exception).

Reply:
Inspecting the code directly, I don't really quite understand what you're saying here. Is the concern that without the extra stack frame from the format call the RecursionError won't trigger? I added this before my changes and it did fail on base 3.9

My intent with creating an exc directly is that it's shared functionality between format_exception, print_exception, etc., so if someone was to, say, refactor format_exception() this would still test that the recursive context is appropriately handled. After all the bug is in the TracebackException constructor not the format()

Reply:
The constructor creates a chain of exceptions, and your change truncated this chain to a length where the constructor's recursion hit its limit. Now format() processes this chain (also recursively). Imagine that for each exception in the chain, format() makes several function calls (so it creates more frames than the constructor). Then it can fail despite the constructor truncating the chain. 

So the test needs to make sure this doesn't happen, by calling the constructor and then calling format().

Reply:
I replaced `TracebackException` with `format_exception`, which exercises both the constructor and format. Is that what you meant? This passed without changes.

I also tried to break it:
```
            exc_info = sys.exc_info()
            e = traceback.TracebackException(exc_info[0], exc_info[1], exc_info[2])
            def weird(e, i):
                if i == 0:
                    list(e.format())
                else:
                    print('weird', i)
                    weird(e, i -1)
            weird(e, 100)
```
The above _still_ passes (I thought `weird(e, 1)` would be enough to break this per your suggestion, since by inspection format() has a recursion that for this test case is equal in depth to that of the constructor. If I was interpreting the `yield from` in format that does that recursion strictly per its [semantics](https://www.python.org/dev/peps/pep-0380/#formal-semantics), then I'd expect `weird(e, 1)` to break.

Is the `yield from` op in the python vm doing some kind of TCO-like inlining? Maybe `yield from f()`.

Anyways, the highest value for which the above works is 483. So seems like catching in the constructor works fine here. Would love to hear an explanation though.